core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,220 @@
+        mDb.newIterator(mBlockLocationsColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seek(Longs.toByteArray(id));
+    List<BlockLocation> locations = new ArrayList<>();
+    for (; iter.isValid(); iter.next()) {
+      byte[] key = iter.key();
+      long workerId =
+          Longs.fromBytes(key[8], key[9], key[10], key[11], key[12], key[13], key[14], key[15]);
+      String tier = new String(iter.value(), Charsets.UTF_8);
+      locations.add(new BlockLocation(workerId, tier));
+    }
+    return locations;
+  }
+
+  @Override
+  public void addLocation(long id, BlockLocation location) {
+    byte[] key = toByteArray(id, location.getWorkerId());
+    try {
+      mDb.put(mBlockLocationsColumn, key, location.getTierAlias().getBytes(Charsets.UTF_8));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeLocation(long blockId, long workerId) {
+    byte[] key = toByteArray(blockId, workerId);
+    try {
+      mDb.delete(mBlockLocationsColumn, key);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<BlockMeta> iterator() {
+    RocksIterator iter =
+        mDb.newIterator(mBlockMetaColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seekToFirst();
+    return new Iterator<BlockMeta>() {
+      @Override
+      public boolean hasNext() {
+        return iter.isValid();
+      }
+
+      @Override
+      public BlockMeta next() {
+        try {
+          return new BlockMeta(Longs.fromByteArray(iter.key()), Longs.fromByteArray(iter.value()));
+        } finally {
+          iter.next();
+        }
+      }
+    };
+  }
+
+  // Implementation based on
+  // https://google.github.io/guava/releases/21.0/api/docs/src-html/com/google/common/primitives/
+  // Longs.html#line.267
+  private byte[] toByteArray(long long1, long long2) {
+    byte[] key = new byte[16];
+    for (int i = 7; i >= 0; i--) {
+      key[i] = (byte) (long1 & 0xffL);
+      long1 >>= 8;
+    }
+    for (int i = 15; i >= 8; i--) {
+      key[i] = (byte) (long2 & 0xffL);
+      long2 >>= 8;
+    }
+    return key;
+  }
+
+  private void initDb(String baseDir) throws RocksDBException {
+    if (mDb != null) {
+      // Column handles must be closed before closing the db, or an exception gets thrown.
+      mDefaultColumn.close();
+      mBlockMetaColumn.close();
+      mBlockLocationsColumn.close();
+      mDb.close();
+      try {
+        FileUtils.deletePathRecursively(mDbPath);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    new File(baseDir).mkdirs();
+
+    ColumnFamilyOptions cfOpts = new ColumnFamilyOptions()
+        .useFixedLengthPrefixExtractor(8); // We always search using the initial long key
+
+    List<ColumnFamilyDescriptor> cfDescriptors = Arrays.asList(
+        new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, cfOpts),
```

Comment:
NIT: can we create the `mBlockMetaColumn` members first, and then add them to the array?